src/App.jsx

```diff
@@ -14,33 +17,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const goodsFromServerWithId = goodsFromServer.map((good, index) => ({
+  name: good,
+  id: index + 1,
+}));
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const ALPHABET_SORT_FIELD = 'alphabet';
+const LENGTH_SORT_FIELD = 'length';
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+function prepareGoodsForOutput(gds, { sortField, reversed }) {
+  const goods = [...gds];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortField === ALPHABET_SORT_FIELD) {
+    goods.sort(({ name: name1 }, { name: name2 }) =>
+      name1.localeCompare(name2),
+    );
+  }
+
+  if (sortField === LENGTH_SORT_FIELD) {
+    goods.sort(
+      ({ name: name1 }, { name: name2 }) => name1.length - name2.length,
+    );
+  }
+
+  if (reversed) {
+    goods.reverse();
+  }
+
+  return goods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+  const preparedGoods = prepareGoodsForOutput(goodsFromServerWithId, {
+    sortField,
+    reversed,
+  });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(ALPHABET_SORT_FIELD)}
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortField !== ALPHABET_SORT_FIELD,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortField(LENGTH_SORT_FIELD)}
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortField !== LENGTH_SORT_FIELD,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => (reversed ? setReversed(false) : setReversed(true))}
```

Comment:
minor: 
```suggestion
          onClick={() => setReversed(!reversed)}
```

or better:
```suggestion
          onClick={() => setReversed(reversed => !reversed)}
```